### PR TITLE
[Fix] `no-unknown-property` support new `precedence` prop

### LINKED ISSUE #3827: [Bug]: Unknown property 'precedence' found (react/no-unknown-property)
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

react v19 introduces a new `precedence` string prop to the [`<link>`](https://react.dev/reference/react-dom/components/link#props) and [`<style>`](https://react.dev/reference/react-dom/components/style#props) elements, but the `react/no-unknown-property` rule doesn’t yet know about it and generates linting errors when used:

```jsx
<>
    <link href="https://foo.bar" precedence="medium" rel="canonical" />
    <style href="unique-hash" precedence="anything">{` p { color: red; } `}</style>
</>
```

example error:
```
  211:39  error  Unknown property 'precedence' found  react/no-unknown-property
```

these occur when running my `lint` npm run script: `"lint": "eslint .",`

### Expected Behavior

i expect the new `precedence` attribute to not be considered an “unknown” property

### eslint-plugin-react version

v7.36.1

### eslint version

v8.57.0

### node version

v22.7.0

### PR DESCRIPTION
**Pull request by acusti**
Fixes #3827

### CODE REVIEW

**Review by ljharb — CHANGES_REQUESTED** [review]
This looks great, except that we should only be supporting this property when the react version is 19+.

**Comment by acusti** [comment]
@ljharb is there a way to accommodate that in the plugin, or does that just mean not adding it until there is a version of the plugin released only for react v19+?

**Comment by ljharb** [comment]
@acusti there is! see the `testReactVersion` function

File: tests/lib/rules/no-unknown-property.js [hunk 1]
```diff
@@ -89,6 +89,7 @@ ruleTester.run('no-unknown-property', rule, {
     { code: '<div onPointerDown={this.onDown} onPointerUp={this.onUp} />' },
     { code: '<input type="checkbox" defaultChecked={this.state.checkbox} />' },
     { code: '<div onTouchStart={this.startAnimation} onTouchEnd={this.stopAnimation} onTouchCancel={this.cancel} onTouchMove={this.move} onMouseMoveCapture={this.capture} onTouchCancelCapture={this.log} />' },
+    { code: '<link precedence="medium" href="https://foo.bar" rel="canonical" />' },
```
Comment (ljharb) [inline tests/lib/rules/no-unknown-property.js:92]: shouldn't this test case only pass when the react version is set to 19+?

File: lib/rules/no-unknown-property.js [hunk 2]
```diff
@@ -234,7 +234,7 @@ const DOM_PROPERTY_NAMES_ONE_WORD = [
   // OpenGraph meta tag attributes
   'property',
   // React specific attributes
-  'ref', 'key', 'children',
+  'ref', 'key', 'children', 'precedence',
```
Comment (ljharb) [inline lib/rules/no-unknown-property.js:237]: i think maybe this needs to be removed here
```suggestion
  'ref', 'key', 'children',
```

Comment (acusti) [reply lib/rules/no-unknown-property.js]: you were faster than me 😅

File: tests/lib/rules/no-unknown-property.js (hunk 1 above)
Comment (acusti) [reply tests/lib/rules/no-unknown-property.js]: definitely. i used `react: { version: '19.0.0-rc.0' }`, but if it makes more sense to do `19.0.0` even though that doesn’t match any released versions, happy to do that.

**Comment by acusti** [comment]
@ljharb now that React v19 stable is released, i can update the version checks and test settings to specify `19.0.0` as the minimum version (rather than the prerelease friendly `19.0.0-0` or release-candidate-compatible `19.0.0-rc.0`). would that be preferable?

**Review by ljharb — APPROVED** [review]
Thanks! I rebased and refactored the function a bit. LGTM assuming tests pass.

**Comment by acusti** [comment]
wonderful! thanks for reviewing, refactoring, and merging it.

**Pull request merged**